datafusion/physical-expr/src/array_expressions.rs

```diff
@@ -2939,6 +3011,43 @@ mod tests {
             as_uint64_array(&arr).expect("failed to initialize function array_length");
 
         assert_eq!(result, &UInt64Array::from(vec![None]));
+
```

Comment:
Rather than extending this module with more rust tests, can you perhaps add SQL level tests using sqllogictest?

That will ensure the functions are usable end to end and that all the connections are in place

Perhaps https://github.com/apache/arrow-datafusion/blob/main/datafusion/sqllogictest/test_files/array.slt

Reply:
I'm not sure how to create an large list in sql file. I think it's why he done the test here? @Weijun-H

Reply:
Yes, I do not know how we could construct the `LargeList` in sqllogictest, instead of an array with more `i32:MAX` by `make_array`.